All_home_works/hw9/task1.py

```diff
@@ -0,0 +1,39 @@
+"""
+Write a function that merges integer from sorted files and returns an iterator
+file1.txt:
+1
+3
+5
+file2.txt:
+2
+4
+6
+list(merge_sorted_files(["file1.txt", "file2.txt"]))
+[1, 2, 3, 4, 5, 6]
+"""
+from pathlib import Path
+from typing import Iterator, List, Union
+
+
+def merge_sorted_files(file_list: List[Union[Path, str]]) -> Iterator:
+    num_generators = [_get_nums_from_file(file) for file in file_list]
+    numbers = [
+        (index, next(number, float("inf")))
+        for index, number in enumerate(num_generators)
+    ]
+    while not all(1 if num[1] == float("inf") else 0 for num in numbers):
```

Comment:
Let's optimize it a bit for cases when we have many files, and files could have different lengths.

Reply:
Can you describe the result of optimization.   Let's try to talk on big O notation.   n -> number of files, k - the average number of numbers in the file.  

What was before, what you get after?

Reply:
I understood my mistake and remade it

Reply:
Let's optimize it a bit for cases when we have many files, and files could have different lengths.